jest/index.d.ts

```diff
@@ -83,6 +83,8 @@ declare namespace jest {
     function useFakeTimers(): typeof jest;
     /** Instructs Jest to use the real versions of the standard timer functions. */
     function useRealTimers(): typeof jest;
+    /** Creates a mock function similar to jest.fn but also tracks calls to object[methodName] */
+    function spyOn(object: any, method: string): typeof jest;
```

Comment:
@asvetliakov you're right about the return type, my mistake.

About the other thing, I see the benefit so you assure is an existing property of that object. Does that take into account custom properties? Because I assume `T extends {}` creates a new Object type.

Anyway, if that'll take that much time, I could always create another PR by that time, but at least if we merge this now spyOn will be usable in ts-jest.